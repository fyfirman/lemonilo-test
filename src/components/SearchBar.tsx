import React from 'react';
import styled from 'styled-components/native';

const Container = styled.View`
  padding: ${({ theme }) => theme.spacing(1)}px ${({ theme }) => theme.spacing(2)}px;
  background-color: ${({ theme }) => theme.palette.primary.main};
`;

const InputContainer = styled.View`
  background-color: #fff;
  border-radius: 4px;
  padding-left: ${({ theme }) => theme.spacing(1)}px;
  padding-right: ${({ theme }) => theme.spacing(1)}px;
`;

const SearchInput = styled.TextInput`
  font-size: 12px;
  height: 36px;
`;

interface SearchBarProps {
  onFocus?: () => any;
}

const SearchBar = ({ onFocus }: SearchBarProps) => {
  return (
    <Container>
      <InputContainer>
        <SearchInput placeholder="Search items, destinations..." onFocus={onFocus} />
      </InputContainer>
    </Container>
  );
};

export default SearchBar;
