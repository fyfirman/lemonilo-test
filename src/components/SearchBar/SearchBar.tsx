import React from 'react';
import styled from 'styled-components/native';

const Container = styled.Pressable`
  padding: ${({ theme }) => theme.spacing(1)}px ${({ theme }) => theme.spacing(2)}px;
  background-color: ${({ theme }) => theme.palette.primary.main};
`;

const InputContainer = styled.View`
  background-color: #fff;
  border-radius: 4px;
  padding-left: ${({ theme }) => theme.spacing(1)}px;
  padding-right: ${({ theme }) => theme.spacing(1)}px;
`;

const SearchInput = styled.TextInput`
  font-size: 12px;
  height: 36px;
`;

interface SearchBarProps {
  onPress?: () => any;
}

const SearchBar = ({ onPress }: SearchBarProps) => {
  return (
    <Container onPress={onPress}>
      <InputContainer>
        <SearchInput placeholder="Search items, destinations..." editable={false} />
      </InputContainer>
    </Container>
  );
};

export default SearchBar;
