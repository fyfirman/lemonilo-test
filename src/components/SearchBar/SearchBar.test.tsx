import React from 'react';
import { renderWithTheme } from '../../__tests__/helpers';
import SearchBar from './SearchBar';
import 'jest-styled-components';

test('should match with snapshot', () => {
  const onPress = jest.fn();
  const tree = renderWithTheme(<SearchBar onPress={onPress} />);

  expect(tree).toMatchSnapshot();
});
