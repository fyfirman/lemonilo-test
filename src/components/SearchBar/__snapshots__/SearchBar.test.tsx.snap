// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#19A0E0",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={false}
      placeholder="Search items, destinations..."
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "fontSize": 12,
            "height": 36,
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
